libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -124,37 +126,46 @@ public UserIdentity generateUserIdentity(
             String userId, String sub, String vot, ContraIndicators contraIndicators)
             throws HttpResponseExceptionWithErrorBody, CredentialParseException,
                     UnrecognisedCiException {
-        List<VcStoreItem> vcStoreItems = dataStore.getItems(userId);
-
+        List<VcStoreItem> vcStoreItems =
+                VcHelper.filterVCBasedOnProfileType(
+                        dataStore.getItems(userId), VectorOfTrust.valueOf(vot).getProfileType());
         List<String> vcJwts = vcStoreItems.stream().map(VcStoreItem::getCredential).toList();
 
         String vtm = configService.getSsmParameter(CORE_VTM_CLAIM);
 
         UserIdentity.UserIdentityBuilder userIdentityBuilder =
                 UserIdentity.builder().vcs(vcJwts).sub(sub).vot(vot).vtm(vtm);
 
-        if (vot.equals(VectorOfTrust.P2.toString())) {
-            final List<VcStoreItem> successfulVCStoreItems =
-                    getSuccessfulVCStoreItems(vcStoreItems);
-            Optional<IdentityClaim> identityClaim = findIdentityClaim(successfulVCStoreItems);
-            identityClaim.ifPresent(userIdentityBuilder::identityClaim);
+        if (VectorOfTrust.valueOf(vot).getProfileType().equals(ProfileType.GPG45)) {
```

Comment:
I think we might be able to change this to `if (VectorOfTrust.valueOf(vot) == P0)`?

If it's an operational profile we do still want to generate the core identity claim (which will probably contain name, DOB and NINO)

The key is that we only use the appropriate VCs to generate them, which you're doing by filtering above :)

Reply:
done.